tasks/main.yml

```diff
@@ -328,13 +328,55 @@
     __mssql_conf_setting_value: "{{ mssql_ip_address }}"
   when: mssql_ip_address is not none
 
-- name: Configure the TCP port setting
+- name: Register the previous tcpport setting
+  when:
+    - mssql_firewall_configure | bool
+    - mssql_firewall_close_previous | bool
+  shell: grep 'tcpport' {{ __mssql_conf_path }} || true
+  changed_when: false
+  register: __mssql_previous_tcp_port
+
+- name: >-
+    Open the {{ '1433' if mssql_tcp_port is none else mssql_tcp_port }} TCP port
```

Comment:
It's probably worth defining the port as a variable rather than having to repeat the expression `{{ '1433' if mssql_tcp_port is none else mssql_tcp_port }}` in several places

Reply:
It did change the behavior, I fixed so that it works everywhere.
One feature of sqlcmd is that sqlcmd does not update mssq.conf if you set tcp port to the default value 1433. It will keep the config clear.
If you set the port to some custom value 1435 and then back to 1433, the config will have `tcpport = 1433` in it.

Reply:
> It did change the behavior, I fixed so that it works everywhere. One feature of sqlcmd is that sqlcmd does not update mssq.conf if you set tcp port to the default value 1433. It will keep the config clear. If you set the port to some custom value 1435 and then back to 1433, the config will have `tcpport = 1433` in it.

ok, sounds good